docs/source/getting-started/tabular-data/data-prep.ipynb

```diff
@@ -0,0 +1,279 @@
+     "text": [
+      "  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current\n",
+      "                                 Dload  Upload   Total   Spent    Left  Speed\n",
+      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
+      "  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0\n",
+      "100 5933k  100 5933k    0     0  1088k      0  0:00:05  0:00:05 --:--:-- 1139k\n",
+      "Archive:  data.zip\n",
+      "  inflating: sample_FR_LFS.csv       \n",
+      "  inflating: __MACOSX/._sample_FR_LFS.csv  \n"
+     ]
+    }
+   ],
+   "source": [
+    "!curl 'https://github.com/opendp/dp-test-datasets/blob/master/data/sample_FR_LFS.csv.zip?raw=true' --location --output data.zip\n",
+    "!unzip data.zip\n",
+    "df = pl.scan_csv(\"sample_FR_LFS.csv\", ignore_errors=True)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 8,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "estimated_max_partition_len = 60_000_000\n",
+    "\n",
+    "context = dp.Context.compositor(\n",
+    "    data=df,\n",
+    "    privacy_unit=dp.unit_of(contributions=36),\n",
+    "    privacy_loss=dp.loss_of(epsilon=1.0),\n",
+    "    split_evenly_over=5,\n",
+    "    margins={\n",
+    "        (\"YEAR\", ): dp.Margin(max_partition_length=estimated_max_partition_len, max_partition_contributions=4),\n",
+    "        (\"ILOSTAT\", ): dp.Margin(max_partition_length=estimated_max_partition_len),\n",
+    "        (): dp.Margin(public_info= \"lengths\",max_partition_length=estimated_max_partition_len, max_num_partitions=1),\n",
+    "    },\n",
+    ")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Limitations of `with_columns`"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The method `with_columns` is used to add or change columns in a data frame. One common use is adding human-readable labels for variables that have numerical coding. For example, the following query computes the number of people in each labor category and then adds the codes' meaning. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/html": [
+       "<div><style>\n",
+       ".dataframe > thead > tr,\n",
+       ".dataframe > tbody > tr {\n",
+       "  text-align: right;\n",
+       "  white-space: pre-wrap;\n",
+       "}\n",
+       "</style>\n",
+       "<small>shape: (4, 2)</small><table border=\"1\" class=\"dataframe\"><thead><tr><th>ILOSTAT</th><th>len</th></tr><tr><td>str</td><td>u32</td></tr></thead><tbody><tr><td>&quot;Laid Off&quot;</td><td>75639</td></tr><tr><td>&quot;NA&quot;</td><td>35922</td></tr><tr><td>&quot;Working for Pa…</td><td>79844</td></tr><tr><td>&quot;Not Working Bu…</td><td>8002</td></tr></tbody></table></div>"
+      ],
+      "text/plain": [
+       "shape: (4, 2)\n",
```

Comment:
Shape of what again? When people are looking at these docs without looking at others, they may not know what these terms may mean.

Reply:
the dataframe? i can add it again, the thought process was originally that if someone's using polars and python, they would know the basics of the dataframe already